### ISSUE TITLE
Webmin available regardless of domain URL

### ISSUE THREAD

**Issue by ced1check** [webmin/webmin#1945]
We have setup webmin on our server on a dedicated port, accessible through an URL like this:

`https://webmin.company.com:port_number`

On this same server we run apache for other HTTPS requests with another domain like this:

`https://service.company.com/`

So I configured webmin with appropriate SSL certificate for webmin.company.com and added this to the config:

`host=webmin.company.com`

If I try the page https://service.company.com:9898, it ends-up with the webmin login page and a certificate error, but it will let me login anyway!?

Is there a way to block any URL but webmin.company.com ?

**Comment by jcameron** [comment]
There's no way to do this currently, but we could implement it as a feature pretty easily!

**Comment by ced1check** [comment]
That would be real nice, thanks for your consideration.

**Comment by jcameron** [comment]
Yeah actually Ilia's solution of using a dedicated IP is best.

We *could* implement this restriction in Webmin, but the user would still be able to connect to the wrong URL - they'd just get an error message.

**Comment by ced1check** [comment]
Thanks for your replies.

An error message, like 403/503 or such would be much better than being able to get webmin version IMO.

Going through apache or a proxy or dedicated IP would require extra certificates to be issued and maintained or even paying extra IP from our ISP.
While webmin is setup to filter source IP and respond to only a dedicated host, it seemed way easier to use that host setup.

Interestingly while source IP filtering is setup, we are still able to obtain the login page from anywhere and actually get webmin version information.

Thanks for your considerations.

```
Detected Webmin

Version:       2.021
Location:      /
CPE:           cpe:/a:webmin:webmin:2.021

Concluded from version/product identification result:
Server: MiniServ/2.021
<title>Login to Webmin</title>
```

**Comment by jcameron** [comment]
Yes, we include the Webmin version in the HTTP response headers. Is this really a risk though? Most other webservers do the same thing ...

**Comment by ced1check** [comment]
We use intruder.io to scan our servers and it reports webmin version number, on all hosts name we use on the dedicated port for webmin.

Not a big deal, but I added the hosts= in configuration file hoping to remove reports for all hosts but the webmin one.

As security experts would say, finding app version numbers is a security risk. I suppose it becomes a risk when and if a vulnerability is found on that version.

By the way, does it report the version from server= configuration line ? If so can I change it to XXX safely or will it be used somewhere, like in version checks for updates?

**Comment by iliajie** [comment]
I didn't notice before that we were displaying MiniServ version in headers for unauthenticated user.

> Not a big deal, 

It may be a big deal as it narrows down the field of attack (helps an attacker).

> A security expert would say finding app version numbers is a security risk I suppose.

Yep, it could be.

> I changed server=XXX in configuration file and it now reports XXX in header, and doesn't appear to impact webmin access.

It will be overwritten on Webmin upgrade.

---

I have added a [PR](https://github.com/webmin/webmin/pull/1948) that addresses this problem.
